release/models/platform/openconfig-platform.yang

```diff
@@ -411,12 +417,58 @@ module openconfig-platform {
 
     leaf location {
       type string;
+      status deprecated;
       description
         "System-supplied description of the location of the
         component within the system. This could be a bay position,
         slot number, socket location, etc. For component types that
         have an explicit slot-id attribute, such as linecards, the
-        system should populate the more specific slot-id.";
+        system should populate the more specific slot-id.
+
+        This leaf is deprecated and replaced by install-position and
+        install-component.";
+    }
+
+    leaf install-position {
+      type string;
+      description
+        "System-supplied index to a position where this component is
+        installed.  The position may be referred in device documenation
+        as a port, slot, bay, socket, etc.  This string must only
+        indicate the name of the position, and not any indication of
+        the name of the parent component within the system.  Instead,
+        parent component name should be present in the 'parent' leaf.
+
+        Typically the install-position is a number, but it is observed
+        that some devices may use letters or alphanumerics.  The
+        position name should be the same name used to physically
+        identify the position in documentation or printed on the
+        device.
+
+        Any component which is removable is expected to have
+        an install-position and an install-component which points to
+        an ancestor component where the connection occurs.
+
+        For component types that have an explicit slot-id attribute,
+        such as LINECARD, the system should populate the more
+        specific slot-id.";
```

Comment:
Likely this part can be removed

Reply:
Thanks for the comment.  I think this is still a relevant and useful hint that install-position should not be used for LINECARD.  

I do think we could open a separate issue to address if we should change `slot-id`.  While slot-id is perhaps redundant to install-position, it does not seem to be in conflict.  For this PR, I want to keep the scope to addressing the 'location' leaf.

Reply:
This made me think we should deprecate slot-id as it is now redundant to install-position and install-component.  I have proposed this for comment.